source/common/router/scoped_config_impl.h

```diff
@@ -132,9 +146,27 @@ class ScopeKeyBuilderImpl : public ScopeKeyBuilderBase {
   std::vector<std::unique_ptr<FragmentBuilderBase>> fragment_builders_;
 };
 
+// ScopedRouteConfiguration and corresponding RouteConfigProvider.
+class ScopedRouteInfo {
+public:
+  ScopedRouteInfo(envoy::api::v2::ScopedRouteConfiguration&& config_proto,
+                  ConfigConstSharedPtr&& route_config);
+
+  ConfigConstSharedPtr routeConfig() const { return route_config_; }
```

Comment:
nit: usually we return `const ConfigConstSharedPtr&`
And leave the caller to decide if a copy(bump ref count) is needed

Reply:
This config is supposed to be shared between workers which do the actually routing, they need to take a share of the ownership of it. 
but I don't have a strong feeling as well.

Reply:
Let me make it clear.
> This config is supposed to be shared between workers which do the actually routing, they need to take a share of the ownership of it.
By `They` I believe you refers the caller of routeConfig(). They could copy if they need. 

Of course it can never be wrong if you do the early copy at the `routeConfig()`

Reply:
returning an reference also means the function makes a promise to callers about the life circle of the return object (personally I dont like returning reference), just imagine what if underlying obj is updated in one thread, and is read in another thread(we are distributing a scopedRouteInfo across threads).
But in this specific case, the ownership will be distributed anyway and the immutability of ScopedRouteInfo is guaranteed by the SRDS subscription, so I agree we can return a reference here.